### ISSUE TITLE
different output for .re & .res object access

### ISSUE THREAD

**Issue by eWert-Online** [melange-re/melange#329]
The following `.res` code
```res
let handleChange = evt => {
  let value: bool = (evt->ReactEvent.Form.target)["checked"]
  onChange(value)
}
```
...compiles to
```js
var handleChange = function (evt) {
  Curry._1(onChange, Caml_oo_curry.js1(321172263, 1, evt.target));
};
```
which produces the following runtime (JavaScript) error message:
```
Uncaught TypeError: Cannot read properties of undefined (reading '0')
    at caml_get_public_method (caml_oo.js:17:7)
    at js1 (caml_oo_curry.js:12:27)
    at handleChange (Switch.bs.js:21:38)
```
---
When writing the same function in `.re` syntax (at least I think it is the same 😄)
```re
let handleChange = evt => {
  let value: bool = evt->ReactEvent.Form.target##checked;
  onChange(value);
};
```
the (correct) output looks like this
```js
var handleChange = function (evt) {
  Curry._1(onChange, evt.target.checked);
};
```
---
`bs-platform@9.0.2` compiles both `.res` and `.re` code snippets to the correct JavaScript output.

**Comment by anmonteiro** [comment]
I thought I addressed this in https://github.com/melange-re/melange/pull/278. Are you running a melange with that commit?

**Comment by eWert-Online** [comment]
I was on commit dd46baf2e1bd8ae8107ac7ca3e1677f8fab658d5 (which also includes this PR). To make sure, I updated to the latest fork commit (43511419eedd42fba11d81698afbc07e047c5bb8) but it didn't change the behaviour.

Should I create a repository with a reproduction example? 🙂

**Comment by anmonteiro** [comment]
Added a fix in #330
